pyaedt/application/Design.py

```diff
@@ -939,6 +721,54 @@ def default_solution_type(self):
         """
         return solutions_defaults[self._design_type]
 
+    @pyaedt_function_handler()
+    def _find_design(self):
+        activedes = None
+        warning_msg = ""
+        names = self.get_oo_name(self.oproject)
+        if names:
+            valids = []
+            for name in names:
+                des = self.get_oo_object(self.oproject, name)
+                if des.GetDesignType() == self.design_type:
+                    if self.design_type in [
+                        "Circuit Design",
+                        "Twin Builder",
+                        "HFSS 3D Layout Design",
+                        "EMIT",
+                        "Q3D Extractor",
+                    ]:
+                        valids.append(name)
+                    elif not self._temp_solution_type:
+                        valids.append(name)
+                    elif self._temp_solution_type in des.GetSolutionType():
+                        valids.append(name)
+            if len(valids) != 1:
+                warning_msg = "No consistent unique design is present. Inserting a new design."
+            else:
+                activedes = valids[0]
+                warning_msg = "Active Design set to {}".format(valids[0])
+        # legacy support for version < 2021.2
+        elif self.design_list:  # pragma: no cover
+            self._odesign = self._oproject.GetDesign(self.design_list[0])
+            if not self._check_design_consistency():
+                count_consistent_designs = 0
+                for des in self.design_list:
+                    self._odesign = self._oproject.SetActiveDesign(des)
+                    if self._check_design_consistency():
+                        count_consistent_designs += 1
+                        activedes = des
+                if count_consistent_designs != 1:
+                    warning_msg = "No consistent unique design is present. Inserting a new design."
+                else:
+                    self.logger.info("Active Design set to {}".format(activedes))
```

Comment:
Nitpick: pyansys docs recommend [Use %-Formatting for Strings
](https://dev.docs.pyansys.com/guidelines/logging.html?highlight=logging#use-formatting-for-strings).

Reply:
I need to check if this syntax is fully compatible with ironpython